### PR TITLE
feat: add `aws_cognito_identity_pool` table

### PR DESCRIPTION
**Pull request by pdecat**
This PR adds the `aws_cognito_identity_pool` table.

~Marked as draft as it is based on the branch of #1854, will rebase on `main` once that one is merged.~

# Integration test logs
<details>
  <summary>Logs</summary>

```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select
  identity_pool_id,
  identity_pool_name,
  tags,
  region,
  account_id
from
  aws_cognito_identity_pool;
+------------------------------------------------+---------------------------+------+-----------+--------------+
| identity_pool_id                               | identity_pool_name        | tags | region    | account_id   |
+------------------------------------------------+---------------------------+------+-----------+--------------+
| eu-west-3:e96205bf-1ef2-4fe6-a748-65e948673960 | test-pdecat-identity-pool | {}   | eu-west-3 | 012345678901 |
+------------------------------------------------+---------------------------+------+-----------+--------------+

> select
  identity_pool_id,
  identity_pool_name,
  allow_classic_flow
from
  aws_cognito_identity_pool
where
  allow_classic_flow = true;
+------------------------------------------------+---------------------------+--------------------+
| identity_pool_id                               | identity_pool_name        | allow_classic_flow |
+------------------------------------------------+---------------------------+--------------------+
| eu-west-3:e96205bf-1ef2-4fe6-a748-65e948673960 | test-pdecat-identity-pool | true               |
+------------------------------------------------+---------------------------+--------------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_cognito_identity_pool.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+			{
+				Name:        "developer_provider_name",
+				Description: "The 'domain' by which Cognito will refer to your users.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "open_id_connect_provider_arns",
+				Description: "The ARNs of the OpenID Connect providers.",
+				Hydrate:     getCognitoIdentityPool,
+				Type:        proto.ColumnType_JSON,
```
Comment (madhushreeray30) [inline aws/table_aws_cognito_identity_pool.go:71]: we need to add a transform function > 
```
Transform:   transform.FromField("OpenIdConnectProviderARNs"),
```

File: aws/table_aws_cognito_identity_pool.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+			},
+			{
+				Name:        "saml_provider_arns",
+				Description: "An array of Amazon Resource Names (ARNs) of the SAML provider for your identity pool.",
+				Hydrate:     getCognitoIdentityPool,
+				Type:        proto.ColumnType_JSON,
```
Comment (madhushreeray30) [inline aws/table_aws_cognito_identity_pool.go:77]: we need to add a transform function > 
```
Transform:   transform.FromField("SamlProviderARNs"),
```

File: aws/table_aws_cognito_identity_pool.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+			},
+			// Steampipe standard columns
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getCognitoIdentityPool,
+				Transform:   transform.FromField("Arn").Transform(transform.EnsureStringArray),
```
Comment (madhushreeray30) [inline aws/table_aws_cognito_identity_pool.go:91]: we do not have a field `Arn`

**Review by madhushreeray30 — COMMENTED** [review]
@pdecat can you please have a look at the comments, thanks!

File: aws/table_aws_cognito_identity_pool.go [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getCognitoIdentityPool,
+				Transform:   transform.FromField("IdentityPoolTags"),
+			},
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
```
[the remark this answers is not in the document]
Comment (pdecat) [reply aws/table_aws_cognito_identity_pool.go]: Done!

Turned out it was `IdentityPoolTags`, not `identity_pool_name`.

File: aws/table_aws_cognito_identity_pool.go (hunk 1 above)
Comment (pdecat) [reply aws/table_aws_cognito_identity_pool.go:71]: Done!

File: aws/table_aws_cognito_identity_pool.go (hunk 2 above)
Comment (pdecat) [reply aws/table_aws_cognito_identity_pool.go:78]: Done!

File: aws/table_aws_cognito_identity_pool.go (hunk 3 above)
Comment (pdecat) [reply aws/table_aws_cognito_identity_pool.go]: Indeed :facepalm: 

Fixed!

**Comment by pdecat** [comment]
Hi @madhushreeray30, sorry I was a bit hasty on this one. I believe I've addressed all comments. PTAL :pray:

**Comment by pdecat** [comment]
After further testing with a more complete Identity Pool test resource, I've discovered one last issue with `developer_provider_name` that I just fixed.

Here's the result of a full query:
```
# steampipe query "select * from aws_cognito_identity_pool;"
+------------------------------------------------+----------------------------------+----------------------+--------------------+---------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------+-----------------------------------------------------------+--------------------------------------------------------------+--------------------------------+-----------------------------------------------------------------------------------------------------------------+-------------------+----------------------+-----------+-----------+--------------+---------------------------+
| identity_pool_id                               | allow_unauthenticated_identities | identity_pool_name   | allow_classic_flow | cognito_identity_providers                                                                                                                        | developer_provider_name  | open_id_connect_provider_arns                             | saml_provider_arns                                           | supported_login_providers      | akas                                                                                                            | tags              | title                | partition | region    | account_id   | _ctx                      |
+------------------------------------------------+----------------------------------+----------------------+--------------------+---------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------+-----------------------------------------------------------+--------------------------------------------------------------+--------------------------------+-----------------------------------------------------------------------------------------------------------------+-------------------+----------------------+-----------+-----------+--------------+---------------------------+
| eu-west-3:8c3fa216-106c-499e-a7da-d8dfc730bd7b | false                            | test-pdecat-id-poool | true               | [{"ClientId":"4rdf5osk3gnf97pt20uhkq1oua","ProviderName":"cognito-idp.eu-west-3.amazonaws.com/eu-west-3_ylM0N2LBF","ServerSideTokenCheck":false}] | test-pdecat-dev-provider | ["arn:aws:iam::012345678901:oidc-provider/iam.clara.net"] | ["arn:aws:iam::012345678901:saml-provider/test-pdecat-saml"] | {"accounts.google.com":"test"} | ["arn:aws:cognito-identity:eu-west-3:012345678901:identitypool/eu-west-3:8c3fa216-106c-499e-a7da-d8dfc730bd7b"] | {"test":"pdecat"} | test-pdecat-id-poool | aws       | eu-west-3 | 012345678901 | {"connection_name":"aws"} |
+------------------------------------------------+----------------------------------+----------------------+--------------------+---------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------+-----------------------------------------------------------+--------------------------------------------------------------+--------------------------------+-----------------------------------------------------------------------------------------------------------------+-------------------+----------------------+-----------+-----------+--------------+---------------------------+

```

**Comment by madhushreeray30** [comment]
@pdecat thanks for the updates. The PR looks good to me 👍

**Review by madhushreeray30 — APPROVED** [review]

**Review by misraved — APPROVED** [review]

**Pull request merged**